### PR TITLE
Supress PHPUnit ReflectionType deprecation warnings on PHP 7.4

### PR DESCRIPTION
**Pull request by garas**
Fix 2 `Trying to access array offset on value of type null` warnings which revealed after suppressing reflection errors.

Supress `ReflectionType` deprecation errors when generating mocks in `TestCase::getMockBuilder() -> MockBuilder::getMock()`, `TestCase::getMockClass()` and `TestCase::getMockForAbstractClass()`.

Plugins, which uses `Cake\TestSuite\TestCase` in their tests, will get suppressed warnings too.

### CODE REVIEW

File: src/Database/Expression/TupleComparison.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ protected function _stringifyValues($generator)
             if ($isMulti) {
                 $bound = [];
                 foreach ($value as $k => $val) {
-                    $valType = $multiType ? $type[$k] : $type;
+                    $valType = $multiType && isset($type[$i]) ? $type[$k] : $type;
```
Comment (markstory) [inline src/Database/Expression/TupleComparison.php]: Why do you check for `$i` when the read is using `$k`?

File: src/TestSuite/MockBuilder.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.8.8
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite;
+
+use PHPUnit\Framework\MockObject\MockBuilder as BaseMockBuilder;
+
+/**
+ * PHPUnit MockBuilder with muted Reflection errors
```
Comment (markstory) [inline src/TestSuite/MockBuilder.php]: Should we flag this as `@internal`?

File: src/Database/Expression/TupleComparison.php (hunk 1 above)
Comment (garas) [reply src/Database/Expression/TupleComparison.php]: Just stupid copy/paste...

File: src/TestSuite/MockBuilder.php (hunk 2 above)
Comment (garas) [reply src/TestSuite/MockBuilder.php]: Done

**Comment by markstory** [comment]
Great work. This is a simple solution compared to the alternatives.

**Comment by dereuromark** [comment]
You might need some shimming for PHP5.6 packages it seems
I have now these fails in 5.6 builds: https://travis-ci.org/dereuromark/cakephp-dto/jobs/630604416

> Fatal error:  Class 'PHPUnit\Framework\MockObject\MockBuilder' not found in /home/travis/build/dereuromark/cakephp-dto/vendor/cakephp/cakephp/src/TestSuite/MockBuilder.php on line 25

**Comment by garas** [comment]
Try to add `loadPHPUnitAliases()`. 

Aliases autoloaded in `FixtureInjector`, `FixtureManager` and `TestSuite`, and `tests/bootstrap.php`. It seems your plugin doesn't use any of these. Adding this to shim `MockBuilder` can be good idea.

**Pull request merged**